src/Hl7.Fhir.Specification/Specification/Snapshot/DifferentialTreeConstructor.cs

```diff
@@ -64,6 +62,9 @@ namespace Hl7.Fhir.Specification.Snapshot
     /// </summary>
     public class DifferentialTreeConstructor
     {
+        /// <summary>Singleton. Global default instance.</summary>
+        public static DifferentialTreeConstructor Default { get; } = new DifferentialTreeConstructor();
```

Comment:
Maybe `MakeTree` could just be a static method instead? Or even an extension for `IEnumerable<ElementDefinition>`

Reply:
Hi Brendan, great suggestion! I added a public extension on `StructureDefinition.DifferentialComponent`, as this method is expected to be called on differentials. The extension method simply calls the original (public) MakeTree() method, which expects a list of `ElementDefinitions`.